3scale/porta
# THREESCALE-10826: Connect to Redis through Unix socket

**What this PR does / why we need it**:

The redis server can be configured to listen in a unix domain file. In this case, the client should provide a URL like:

```
unix:///var/run/redis.sock
```

However our Redis config parser doesn't accept `unix://` URLs. We could fix that, and we might do eventually, but this PR adds some new environment variables, `REDIS_PATH` and `BACKEND_REDIS_PATH`, in order to provide the unix socket explicitly with it's own parameter `path`. This is how Sidekiq and our Redis gem expects the socket to be given.

**Which issue(s) this PR fixes** 

This is part of https://issues.redhat.com/browse/THREESCALE-10826

**Verification steps** 

1. Launch a redis container listening in a host local file with this command
```
podman run --name redis-unix -v /path/to/your/run:/var/run/:z redis:6.2-alpine redis-server --unixsocket /var/run/redis.sock --unixsocketperm 777 --port 0
```
2. Set `REDIS_PATH` and `BACKEND_REDIS_PATH` to `/path/to/your/run/redis.sock`
3. Sidekiq and porta should work as usual.

Additionally, ACL credentials can be provided with `[BACKEND_]REDIS_USERNAME` and `[BACKEND_]REDIS_PASSWORD` if the server is authenticated. Sentinels and TLS connections cannot be established through unix sockets, though.
